posthog/models/person/util.py

```diff
@@ -148,25 +149,39 @@ def get_persons_by_uuids(team: Team, uuids: List[str]) -> QuerySet:
     return Person.objects.filter(team_id=team.pk, uuid__in=uuids)
 
 
-def delete_person(
-    person_id: UUID, properties: Dict, is_identified: bool, delete_events: bool = False, team_id: int = False
-) -> None:
+def delete_person(person: Person, delete_events: bool = False, delete_distinct_ids=False) -> None:
```

Comment:
I'm feeling iffy about encoding as much business logic into this as we are.

This is only called in a single callsite (tests don't count) - let's create a separate function for deleting distinct ids instead.

Reply:
done